### PR TITLE
[Version] Only strip leading "v" or "version-" when parsing SemVer

### PR DESCRIPTION
**Pull request by segiddins**
This means you can have arbitrary, non-version tags that _happen_ to end in numbers, and those trailing numbers will no longer be interpreted as a semantic version.

### CODE REVIEW

**Comment by neilpa** [comment]
This is probably too strict a requirement (c.f. #428). Can this instead require a `digit . digit` prefix? That should fix most of the false positives that match a single number

**Comment by jspahrsummers** [comment]
Thanks for the pull request, @segiddins! :sparkles:

I agree with @neilpa's feedback, and think the requirement for two version components makes a lot of sense.

**Comment by segiddins** [comment]
Should that check be done in `SemanticVersion.fromScanner` or here in `fromPinnedVersion` ?

**Comment by segiddins** [comment]
(Also, reading #428, it seems like that issue was requesting something more like this?)

**Comment by neilpa** [comment]
`fromScanner` should require a minor version component. But that also requires `fromPinnedVersion` to keep scanning if in case of tags like `RAC2-2.5.1`

**Comment by neilpa** [comment]
RE #428 - More-or-less, but I didn't have strong feelings for any specific solution when I filed that.

**Comment by segiddins** [comment]
I'm happy to implement that, but it seems like then you're just moving the restriction from `foo-\d` to `foo-\d.\d`?

**Comment by neilpa** [comment]
Yea, but in practice that should solve most of the false identification.

**Comment by segiddins** [comment]
Making the change to require a minor version in `fromScanner` makes an existing test fail -- should I just remove that test? (it's for `v1`)

**Comment by neilpa** [comment]
Or you could make it expect to fail like with the other test you added.

**Comment by segiddins** [comment]
Amended.

File: Source/CarthageKitTests/VersionSpec.swift [hunk 1]
```diff
@@ -31,9 +31,10 @@ class SemanticVersionSpec: QuickSpec {
 
 		it("should parse semantic versions") {
 			expect(SemanticVersion.fromPinnedVersion(PinnedVersion("1.4")).value).to(equal(SemanticVersion(major: 1, minor: 4, patch: 0)))
-			expect(SemanticVersion.fromPinnedVersion(PinnedVersion("v1")).value).to(equal(SemanticVersion(major: 1, minor: 0, patch: 0)))
+			expect(SemanticVersion.fromPinnedVersion(PinnedVersion("v1")).value).to(beNil())
 			expect(SemanticVersion.fromPinnedVersion(PinnedVersion("v2.8.9")).value).to(equal(SemanticVersion(major: 2, minor: 8, patch: 9)))
 			expect(SemanticVersion.fromPinnedVersion(PinnedVersion("v2.8-alpha")).value).to(beNil())
+			expect(SemanticVersion.fromPinnedVersion(PinnedVersion("null-string-beta-2")).value).to(beNil())
```
Comment (neilpa) [inline Source/CarthageKitTests/VersionSpec.swift]: Can you move these to a new spec for testing failed parsings

Comment (segiddins) [inline Source/CarthageKitTests/VersionSpec.swift]: done!

**Comment by neilpa** [comment]
:+1: 

Note that this will still fail on `good2go-1.2.3`. Although, I'm okay with that for now since this would've failed in the same way before this change.

**Comment by jdhealy** [comment]
Hmm, I almost just tagged something with `swiftlang-700.0.38.1`, which gets interpreted as a version.

Would disallowing tags with more than three dot-joined digit sets be too strict? :confused:

**Comment by jspahrsummers** [comment]
@jdhealy No, that seems appropriate, but let's please discuss it in a new issue or PR.

**Pull request merged**